From ab2d99bd5f6bda8bb66a7b0f1b2f7a75db2800ee Mon Sep 17 00:00:00 2001
From: Samuel Holland <samuel@sholland.org>
Date: Sun, 24 Oct 2021 03:37:12 -0500
Subject: [PATCH] mfd: rk808: Fix power key polarity

The key up and key down events were backwards, as seen by evtest.

Signed-off-by: Samuel Holland <samuel@sholland.org>
Signed-off-by: Peter Geis <pgwipeout@gmail.com>
(cherry picked from commit e8ab688572909126f11f7eba4e7feae9713d9ba9)
---
 drivers/mfd/rk808.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/mfd/rk808.c b/drivers/mfd/rk808.c
index f44fc3f080a8e..67ca7785c2bb3 100644
--- a/drivers/mfd/rk808.c
+++ b/drivers/mfd/rk808.c
@@ -122,13 +122,13 @@ static const struct resource rk817_rtc_resources[] = {
 };
 
 static const struct resource rk805_key_resources[] = {
-	DEFINE_RES_IRQ(RK805_IRQ_PWRON_RISE),
 	DEFINE_RES_IRQ(RK805_IRQ_PWRON_FALL),
+	DEFINE_RES_IRQ(RK805_IRQ_PWRON_RISE),
 };
 
 static const struct resource rk817_pwrkey_resources[] = {
-	DEFINE_RES_IRQ(RK817_IRQ_PWRON_RISE),
 	DEFINE_RES_IRQ(RK817_IRQ_PWRON_FALL),
+	DEFINE_RES_IRQ(RK817_IRQ_PWRON_RISE),
 };
 
 static const struct resource rk817_charger_resources[] = {
-- 
2.38.0

